### PR TITLE
73880 initial project

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 73880: initial project](https://edu-redmine.sun-asterisk.vn/issues/73880)
## WHAT
- Apply MVC structure: Task #73881
- Setup launch screen: Task #73882
- Setup Tabbar: Task #73883
## Evidence (Screenshot or Video)
Screenshot 2024-03-20 at 15 55 21

## Review C
Screenshot 2024-03-20 at 15 55 21
hecklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: .DS_Store
Comment (dongtg-1485) [inline .DS_Store:1]: cho vào `.gitignore`

File: Weather_App/Weather_App/App/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+//  Created by ho.bao.an on 20/03/2024.
+//
+
+import UIKit
+
+@main
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+
+
```
Comment (dongtg-1485) [inline Weather_App/Weather_App/App/AppDelegate.swift]: remove trash lines

File: Weather_App/Weather_App/App/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        // Override point for customization after application launch.
+        Thread.sleep(forTimeInterval: 3)
```
Comment (dongtg-1485) [inline Weather_App/Weather_App/App/AppDelegate.swift]: Sao đoàn này lại block thread 3s vậy em ?

File: Weather_App/Weather_App/App/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
+    }
+
+    func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
+        // Called when the user discards a scene session.
+        // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
+        // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
+    }
+
+
```
Comment (dongtg-1485) [inline Weather_App/Weather_App/App/AppDelegate.swift]: Các function không dùng thì xóa

File: Weather_App/Weather_App/App/SceneDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        // Use this method to undo the changes made on entering the background.
+    }
+
+    func sceneDidEnterBackground(_ scene: UIScene) {
+        // Called as the scene transitions from the foreground to the background.
+        // Use this method to save data, release shared resources, and store enough scene-specific state information
+        // to restore the scene back to its current state.
+    }
+
+
```
Comment (dongtg-1485) [inline Weather_App/Weather_App/App/SceneDelegate.swift]: Các function không dùng thì xóa

File: Weather_App/Weather_App/Ultilities/Assets.xcassets/AccentColor.colorset/Contents.json
Comment (dongtg-1485) [inline Weather_App/Weather_App/Ultilities/Assets.xcassets/AccentColor.colorset/Contents.json:1]: Tách color ra một file Asset riêng

File: Weather_App/Weather_App/Ultilities/Assets.xcassets/AppIcon.appiconset/iconApp.png
Comment (dongtg-1485) [inline Weather_App/Weather_App/Ultilities/Assets.xcassets/AppIcon.appiconset/iconApp.png:1]: Dùng ảnh PDF nếu có thể

File: .DS_Store
Comment (daolq-2712) [inline .DS_Store:1]: Không push file này lên.

Cho vào .gitignore

ref:https://stackoverflow.com/questions/1139762/ignore-files-that-have-already-been-committed-to-a-git-repository

File: .github/.DS_Store
Comment (daolq-2712) [inline .github/.DS_Store:1]: file này cũng vậy

File: .gitignore [hunk 5]
```diff
@@ -87,4 +87,5 @@ fastlane/test_output
 # After new code Injection tools there's a generated folder /iOSInjectionProject
 # https://github.com/johnno1962/injectionforxcode
 
+.DS_Store
```
Comment (daolq-2712) [inline .gitignore:90]:
```suggestion
.DS_Store
.github/.DS_Store
```

File: Weather_App/Weather_App/Scenes/Favorite/FavoriteViewController.swift [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destination.
+        // Pass the selected object to the new view controller.
+    }
+    */
+
```
Comment (daolq-2712) [inline Weather_App/Weather_App/Scenes/Favorite/FavoriteViewController.swift]: remove

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import UIKit
+
+class MapViewController: UIViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+    }
+
+
```
Comment (daolq-2712) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]:
```suggestion
```

File: Weather_App/Weather_App/Scenes/Weather Detail/WeatherDetailViewController.swift [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destination.
+        // Pass the selected object to the new view controller.
+    }
+    */
```
Comment (daolq-2712) [inline Weather_App/Weather_App/Scenes/Weather Detail/WeatherDetailViewController.swift]:
```suggestion
```

File: Weather_App/Weather_App/Scenes/Favorite/FavoriteViewController.swift [hunk 9]
```diff
@@ -0,0 +1,29 @@
+//
+//  FavoriteViewController.swift
+//  Weather_App
+//
+//  Created by ho.bao.an on 20/03/2024.
+//
+
+import UIKit
+
+class FavoriteViewController: UIViewController {
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Favorite/FavoriteViewController.swift]: thêm final e

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 10]
```diff
@@ -0,0 +1,19 @@
+//
+//  MapViewController.swift
+//  Weather_App
+//
+//  Created by ho.bao.an on 20/03/2024.
+//
+
+import UIKit
+
+class MapViewController: UIViewController {
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]: thêm final

File: Weather_App/Weather_App/Scenes/Weather Detail/WeatherDetailViewController.swift [hunk 11]
```diff
@@ -0,0 +1,29 @@
+//
+//  WeatherDetailViewController.swift
+//  Weather_App
+//
+//  Created by ho.bao.an on 20/03/2024.
+//
+
+import UIKit
+
+class WeatherDetailViewController: UIViewController {
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Weather Detail/WeatherDetailViewController.swift]: tương tự mấy thằng khác nhé

**Review by dongtg-1485 — APPROVED** [review]

**Review by daolq-2712 — APPROVED** [review]
LGTM!

**Pull request merged**